pkg/actuators/machine/vm.go

```diff
@@ -105,39 +106,59 @@ func validateVMConfig(mscp *machineScope) field.ErrorList {
 }
 
 func validateClusterConfig(mscp *machineScope, fldPath *field.Path) *field.Error {
-	var err error
 	var errMsg string
+	var clusterName string
 
 	switch mscp.providerSpecValidated.Cluster.Type {
-	case machinev1.NutanixIdentifierName:
+	case configv1.NutanixIdentifierName:
 		if mscp.providerSpecValidated.Cluster.Name == nil || *mscp.providerSpecValidated.Cluster.Name == "" {
 			return field.Required(fldPath.Child("cluster", "name"), "Missing cluster name")
 		} else {
-			clusterName := *mscp.providerSpecValidated.Cluster.Name
+			clusterName = *mscp.providerSpecValidated.Cluster.Name
 			clusterRefUuidPtr, err := findClusterUuidByName(mscp.nutanixClient, clusterName)
 			if err != nil {
 				errMsg = fmt.Sprintf("Failed to find cluster with name %q. error: %v", clusterName, err)
 				return field.Invalid(fldPath.Child("cluster", "name"), clusterName, errMsg)
 			} else {
-				mscp.providerSpecValidated.Cluster.Type = machinev1.NutanixIdentifierUUID
+				mscp.providerSpecValidated.Cluster.Type = configv1.NutanixIdentifierUUID
 				mscp.providerSpecValidated.Cluster.UUID = clusterRefUuidPtr
 			}
 		}
-	case machinev1.NutanixIdentifierUUID:
+	case configv1.NutanixIdentifierUUID:
 		if mscp.providerSpecValidated.Cluster.UUID == nil || *mscp.providerSpecValidated.Cluster.UUID == "" {
 			return field.Required(fldPath.Child("cluster", "uuid"), "Missing cluster uuid")
 		} else {
 			clusterUUID := *mscp.providerSpecValidated.Cluster.UUID
-			if _, err = mscp.nutanixClient.V3.GetCluster(clusterUUID); err != nil {
+			if res, err := mscp.nutanixClient.V3.GetCluster(clusterUUID); err != nil {
 				errMsg = fmt.Sprintf("Failed to find cluster with uuid %v. error: %v", clusterUUID, err)
 				return field.Invalid(fldPath.Child("cluster", "uuid"), clusterUUID, errMsg)
+			} else {
+				clusterName = *res.Spec.Name
 			}
 		}
 	default:
-		errMsg = fmt.Sprintf("Invalid cluster identifier type, valid types are: %q, %q.", machinev1.NutanixIdentifierName, machinev1.NutanixIdentifierUUID)
+		errMsg = fmt.Sprintf("Invalid cluster identifier type, valid types are: %q, %q.", configv1.NutanixIdentifierName, configv1.NutanixIdentifierUUID)
 		return field.Invalid(fldPath.Child("cluster", "type"), mscp.providerSpecValidated.Cluster.Type, errMsg)
 	}
 
+	// If the providerSpec configures the failureDomain reference, validate that the cluster
+	// configuration in both the providerSpec and the referenced failureDomain are consistent.
+	if mscp.failureDomain != nil {
+		fd := mscp.failureDomain
+		switch fd.Cluster.Type {
+		case configv1.NutanixIdentifierName:
+			if *fd.Cluster.Name != clusterName {
+				errMsg = fmt.Sprintf("The cluster configured in the providerSpec is not consistent with that configured in the failureDomain %q: %q", fd.Name, *fd.Cluster.Name)
```

Comment:
i wonder if we could make this error a little clearer, eg "The cluster name specified in the providerSpec is not the same as the cluster name configured in the failureDomain". 

i just want to make sure that the user knows what to look for when debugging this error.

Reply:
changed.